src/Microsoft.Health.Dicom.Core/Features/Common/ParallelEnumerable.cs

```diff
@@ -0,0 +1,127 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Core.Features.Common;
+
+internal sealed class ParallelEnumerationOptions
+{
+    public int MaxBufferedItems { get; init; } = 100;
+
+    public int MaxDegreeOfParallelism { get; init; } = Environment.ProcessorCount * 4;
```

Comment:
These values were specified in Configs/RetrieveConfiguration.cs; just wondering why they're here as well?

Reply:
There are two because they exist at different "levels." One is a method-level set of options while the other exists for the service. Today they have the same settings, but that may change.